modules/internal/ChapelArray.chpl

```diff
@@ -3577,9 +3577,13 @@ module ChapelArray {
   // B would just be initialized to the result of the function call -
   // meaning that B would not refer to distinct array elements.
   pragma "unalias fn"
-  inline proc chpl__unalias(ref x: domain) {
+  inline proc chpl__unalias(x: domain) {
     if x._unowned {
-      chpl_replaceWithDeepCopy(x);
+      pragma "no auto destroy" var ret = x;
```

Comment:
If the other cases continue to call chpl__autoDestroy, I'd put a comment here saying that we could call chpl__autoDestroy but it doesn't matter because it doesn't do anything for an unowned domain. (Or I'd consider adding it anyway).